lib/core/scheduler.rb

```diff
@@ -148,6 +148,21 @@ def configuration_triggered?(connector_settings)
     end
 
     def filtering_validation_triggered?(connector_settings)
+      features = connector_settings.features
+
+      if features.nil? || features.empty?
+        # Regression bug
+        Utility::Logger.error("#{connector_settings.formatted} features are empty. Skip filtering validation.")
```

Comment:
This seems not necessary unless you want to log the error.

What does it imply if it's empty? A migration failed?

Reply:
For example a migration failed or someone tempered with the features manually

Reply:
Maybe we could make the message more explicit like: "... features are empty. This is likely to be an index related error. Please check your index for connector id '...'.)" I just think it would be good to be explicit that this is an error and something which shouldn't happen (as it already happened before).

Reply:
But not a super strong opinion, I can also take it out, if you would prefer it.

Reply:
I do want to take it out. My point is, we should make our code robust, i.e. it won't crash because some of the system indices (`.elastic-connectors`, `.elastic-connector-sync-jobs`) are tempered. 

But I think it's overkill if we check the existence and validity of each field of system indices before use.

Reply:
With the current implementation we won't crash as they'll just be empty if the features are null in the index, right? 

```
def features
      self[:features] || {}
end
```

But if you think it's overkill to check the existence/validity you would advocate for taking out the explicit guarding/logging, right? Sorry, if I'm not understanding fully what's the action to take 😄